modules/collect/src/main/java/com/opengamma/strata/collect/io/CsvFile.java

```diff
@@ -88,7 +90,34 @@ public static CsvFile of(CharSource source, boolean headerRow) {
    */
   public static CsvFile of(CharSource source, boolean headerRow, char separator) {
     ArgChecker.notNull(source, "source");
-    ImmutableList<String> lines = Unchecked.wrap(() -> source.readLines());
+    List<String> lines = Unchecked.wrap(() -> source.readLines());
+    return create(lines, headerRow, separator);
+  }
+
+  /**
+   * Parses the specified reader as a CSV file where the separator is specified and might not be a comma.
+   * <p>
+   * This factory method takes a {@link Reader}.
+   * Callers are encouraged to use {@link CharSource} instead of {@code Reader} where possible.
+   * <p>
+   * This factory method allows the separator to be controlled.
+   * For example, a tab-separated file is very similar to a CSV file, the only difference is the separator.
+   * 
+   * @param reader  the file resource
+   * @param headerRow  whether the source has a header row, an empty source must still contain the header
+   * @param separator  the separator used to separate each field, typically a comma, but a tab is sometimes used
+   * @return the TSV file
+   * @throws UncheckedIOException if an IO exception occurs
+   * @throws IllegalArgumentException if the file cannot be parsed
+   */
+  public static CsvFile of(Reader reader, boolean headerRow, char separator) {
```

Comment:
Same as the comment on `CsvIterator` - would be good to have a `Reader` version of both factory methods for better usability and symmetry.